src/browser/uuid_availability.ts

```diff
@@ -0,0 +1,22 @@
+const electron = require('electron');
+const namedMutex = electron.namedMutex;
+import { meshEnabled } from './connection_manager';
+import { getAppRunningState } from './core_state';
+
+const makeMutexKey = (uuid: string) => `uuid-${uuid}`;
+
+export function isUuidAvailable(uuid: string) {
+    return !getAppRunningState(uuid) && (
+        !meshEnabled || (
+            namedMutex.tryLock(makeMutexKey(uuid)) === 0
```

Comment:
i just looked at the implementation of this `tryLock` function in the Runtime.
it appears this is simply returning the raw value of the OS wait call. (@anguslim confirm?)

be-warned, value of `WAIT_OBJECT_0` (0) *and* `WAIT_ABANDONED` (128) indicates that the caller now owns the mutex.